### PR TITLE
Document InternedStr

### PR DESCRIPTION
**Pull request by jyn514**
r? @Kixiron

### CODE REVIEW

File: src/intern.rs [hunk 1]
```diff
@@ -4,6 +4,9 @@ use std::sync::RwLock;
 use lasso::{Rodeo, Spur};
 use lazy_static::lazy_static;
 
+/// A opaque identifier for a string which has been [interned].
```
Comment (Kixiron) [inline src/intern.rs:7]: Maybe go into a little more detail on _why_ strings are being interned, someone not familiar may not understand the reasoning at a glance

File: src/intern.rs [hunk 2]
```diff
@@ -24,12 +39,28 @@ macro_rules! get_str {
...
+    /// Convert this identifier back into the original `String`, cloning it along the way.
+    ///
+    /// # See also
+    /// [`get_str!`](../macro.get_str.html)
+    ///
+    /// # Panics
+    /// This function will panic if another thread panicked while accessing the global string pool.
+    ///
     pub fn resolve_and_clone(self) -> String {
         get_str!(self).to_string()
```
Comment (Kixiron) [inline src/intern.rs:55]: Not necessarily part of the change, but why is `.to_string()` used when they function is about cloning? They’re functionally the same, but it seems odd

Comment (jyn514) [reply src/intern.rs:55]: `get_str!` returns a `&str`, not a `String`.

Comment (Kixiron) [reply src/intern.rs:60]: Ah, indeed it does, I always forget that clone on `&str` doesn’t do anything

**Pull request merged**